src/pages/storage/HostPathDeviceModal.tsx

```diff
@@ -0,0 +1,98 @@
+import { FC, ReactNode, useRef, useState } from "react";
+import { Button, Input, Modal } from "@canonical/react-components";
+import { InstanceAndProfileFormikProps } from "components/forms/instanceAndProfileFormValues";
+import { LxdDiskDevice } from "types/device";
+
+interface Props {
+  formik: InstanceAndProfileFormikProps;
+  onFinish: (device: LxdDiskDevice) => void;
+  onCancel: () => void;
+  title?: ReactNode;
+}
+
+const HostPathDeviceModal: FC<Props> = ({
+  formik,
+  onFinish,
+  onCancel,
+  title,
+}) => {
+  const [source, setSource] = useState("");
+  const [path, setPath] = useState<string>();
+  const touchedRef = useRef({
+    source: false,
+    path: false,
+  });
+
+  const handleFinish = () => {
+    const device: LxdDiskDevice = {
+      type: "disk",
+      source,
+      path,
+    };
+
+    onFinish(device);
+  };
+
+  return (
+    <Modal
+      className="host-path-device-modal"
+      close={onCancel}
+      title={title}
+      buttonRow={
+        <>
+          <Button
+            appearance="base"
+            className="u-no-margin--bottom"
+            type="button"
+            onClick={onCancel}
+          >
+            Cancel
```

Comment:
```suggestion
            Back
```
I think this should say back, and we should also change the caption in the volume case, which has possibly 3 steps deep with volume creation. Because on cancel, I'd expect the modal to close, but we are going back to the previous step.

Reply:
I think we should distinguish in the logic between this back button and the x of the modal. The x should close it directly, this back button is a way to got to the previous step.